### PR TITLE
Use hash instead of source name to find FB message sink

### PR DESCRIPTION
**Pull request by SkyToGround**
### Issue

* Closes DM-1129

### Description of work

* Replaced the use of the source name to identify a message sink with a hash of the Flatbuffer ID plus source name.
* Fixed relevant unit tests.
* Added unit test.
* Changed GTest to be linked statically to work with CLion on MacOSX.
* Some minor clang-format related changes/improvements.

It is not strictly necessary to use a hash, just using the strings would be good enough in this case. The performance benefit of using a hash are minimal to non-existent. However, I would argue that a hash is a slightly cleaner design that simplifies (potential) future re-designs.

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

File: src/KafkaW/ConfigureKafka.cpp [hunk 1]
```diff
@@ -17,4 +17,4 @@ void configureKafka(RdKafka::Conf *RdKafkaConfiguration,
     }
   }
 }
-}
\ No newline at end of file
+} // namespace KafkaW
```
Comment (SkyToGround) [inline src/KafkaW/ConfigureKafka.cpp]:
```suggestion
} // namespace KafkaW

```

File: src/DemuxTopic.cpp [hunk 2]
```diff
@@ -23,19 +23,19 @@ ProcessMessageResult
...
   try {
-    auto &CurrentSource = TopicSources.at(Message.getSourceName());
+    auto &CurrentSource = TopicSources.at(Message.getSourceHash());
     auto ProcessingResult = CurrentSource.process_message(Message);
     ++messages_processed;
     return ProcessingResult;
   } catch (std::out_of_range &e) {
-    Logger->trace("Source with name \"{}\" is not in list.",
-                  Message.getSourceName());
+    Logger->trace("Source with name \"{}\" and ID \"{}\" is not in list.",
```
Comment (matthew-d-jones) [inline src/DemuxTopic.cpp]: Braces don't need to be escaped.

File: src/Streamer.cpp [hunk 3]
```diff
@@ -145,10 +145,9 @@ FileWriter::Streamer::pollAndProcess(FileWriter::DemuxTopic &MessageProcessor) {
     return ProcessMessageResult::ERR;
   }
 
-  if (std::find(Sources.begin(), Sources.end(), Message->getSourceName()) ==
-      Sources.end()) {
-    Logger->warn("Message from topic \"{}\" has an unknown source name "
-                 "(\"{}\"), ignoring.",
+  if (Sources.find(Message->getSourceHash()) == Sources.end()) {
+    Logger->warn("Message from topic \"{}\" with the source name \"{}\" is "
```
Comment (matthew-d-jones) [inline src/Streamer.cpp]: Braces don't need to be escaped.

File: src/DemuxTopic.cpp (hunk 2 above)
Comment (SkyToGround) [reply src/DemuxTopic.cpp]: I don't understand, I am not escaping braces here. I am escaping the quotation marks.

File: src/Streamer.cpp (hunk 3 above)
Comment (SkyToGround) [reply src/Streamer.cpp]: Ditto.

File: src/DemuxTopic.cpp (hunk 2 above)
Comment (matthew-d-jones) [reply src/DemuxTopic.cpp]: Oops, somehow missed that ^^

Comment (SkyToGround) [reply src/DemuxTopic.cpp]: np

**Comment by SkyToGround** [comment]
I will merge when Jenkins is done then.

**Pull request merged**